### PR TITLE
packer: depend on gox at build time

### PR DESCRIPTION
**Pull request by commitay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
https://github.com/Homebrew/homebrew-core/pull/22005, https://github.com/Homebrew/homebrew-core/issues/21801

### CODE REVIEW

File: Formula/packer.rb [hunk 1]
```diff
@@ -16,34 +14,16 @@ class Packer < Formula
...
     packerpath.install Dir["{*,.git}"]
-    Language::Go.stage_deps resources, buildpath/"src"
-
-    cd "src/github.com/mitchellh/gox" do
-      system "go", "build"
-      buildpath.install "gox"
-    end
 
     cd packerpath do
       # We handle this step above. Don't repeat it.
```
Comment (ilovezfs) [inline Formula/packer.rb]: This comment seems like it needs to be revised

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/packer.rb [hunk 2]
```diff
@@ -16,34 +14,16 @@ class Packer < Formula
   end
 
   depends_on "go" => :build
+  depends_on "gox" => :build
```
Comment (ilovezfs) [inline Formula/packer.rb]: build deps should be alphabetized amongst themselves

File: Formula/packer.rb (hunk 1 above)
Comment (commitay) [reply Formula/packer.rb]: `These are provided by the gox formula and upstream vendoring`?

Comment (ilovezfs) [reply Formula/packer.rb]: I don't know why we don't need stringer vendored. I guess the build never actually uses it?

Maybe just
```
# Avoid running `go get`
```
and leave the rest to our imagination

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
:package: Thanks @commitay!

**Pull request closed without merging**